sim-lib/src/lib.rs

```diff
@@ -418,16 +418,58 @@ async fn consume_simulation_results(
     shutdown: triggered::Trigger,
 ) {
     log::debug!("Simulation results consumer started.");
+    let mut result_logger = PaymentResultLogger::new();
 
     while let Some(resolved_payment) = receiver.recv().await {
-        // TODO - write to CSV.
-        println!("Resolved payment received: {:?}", resolved_payment);
+        result_logger.report_result(&resolved_payment.0, &resolved_payment.1);
+        log::trace!("Resolved payment received: {:?}", resolved_payment);
     }
 
     log::debug!("Simulation results consumer exiting");
     shutdown.trigger();
 }
 
+/// PaymentResultLogger is an aggregate logger that will report on a summary of the payments that have been reported
+/// to it at regular itervals (defined by the log_interval it is created with).
+#[derive(Default)]
+struct PaymentResultLogger {
+    success_payment: u64,
+    failed_payment: u64,
+    total_sent: u64,
+    call_count: u8,
+    log_interval: u8,
+}
+
+impl PaymentResultLogger {
+    fn new() -> Self {
+        PaymentResultLogger {
+            // TODO: set the interval at which we log based on the number of payment we're expecting to log.
+            log_interval: 10,
+            ..Default::default()
+        }
+    }
+
+    fn report_result(&mut self, details: &DispatchedPayment, result: &PaymentResult) {
+        match result.failure_reason {
+            Some(_) => self.failed_payment += 1,
+            None => self.success_payment += 1,
+        }
```

Comment:
I hate empty `Some`

```suggestion
        if result.failure_reason.is_some() {
            self.failed_payment += 1
        } else {
            self.success_payment += 1
        }
```

Reply:
match is _one_ less LOC so I'm going to leave this one :')

Reply:
Fair enough